### DISCUSSION TITLE
how to use env i vscode extension api

### DISCUSSION THREAD

**Discussion by iamalbinnj** [microsoft/vscode-discussions#1073] [Extension Development QnA] [answered]
am creating vscode extension so i need hide token so i need to use env...so is there any way

**Comment by alefragnani** [comment, marked as answer]
Hi @iamalbinnj ,

If you need to store tokens, I think you should use the `SecretStorage API` (https://code.visualstudio.com/api/references/vscode-api#SecretStorage), as it was created with this in mind. BTW, last year this API was updated to use Electron's safeStorage API over keytar (https://code.visualstudio.com/updates/v1_80#_secretstorage-api-now-uses-electron-api-over-keytar).

But, if you really need to use `env`, you could try using `process.env` from Node.js.

Hope this helps

**Reply by iamalbinnj** [reply]
thanks for more information, but if i use env ,where to store env variables
